core/state/state_object.go

```diff
@@ -157,11 +157,20 @@ func (s *stateObject) touch() {
 
 func (s *stateObject) getTrie(db Database) Trie {
 	if s.trie == nil {
-		var err error
-		s.trie, err = db.OpenStorageTrie(s.addrHash, s.data.Root)
-		if err != nil {
-			s.trie, _ = db.OpenStorageTrie(s.addrHash, common.Hash{})
-			s.setError(fmt.Errorf("can't create storage trie: %v", err))
+		// Try fetching from prefetcher first
```

Comment:
If I understand correctly, whenever we start to process block:

- Resume the `triePrefetcher`, start a new cycle
- Do whatever change, load state from cache/snapshot/database
- Finalise by transaction, emit tasks from path resolving
- Pause the `triePerfetcher`, end the cycle
- Provide resolved trie for rehashing

But there during the transaction executing, is all tries unavailable from the `triePrefetcher`?

Reply:
I think you understood correctly, but the middle sections happens once per tx. So after each tx, we ship stuff to the prefetcher, so that it can resolve the paths and preload the tries. After all txs are done, we can do the commit operation on the already resolved tries. 

So if I reformulate your question the way I interpret it: During tx execution, can we use any of the work that the `triePrefetcher` does? 

Answer: Yes and no.  

Yes: We can't access the tries, but the prefetcher loads them up into the clean-cache. So the actual trie blobs become 'hot'. The first version of this PR did only that, and this more mature varaint also delivers the actual tries in object form. 

No: We don't actually have any use for the tries, until we do a commit operation (assuming we are snapshot-backed). 

Maybe I didn't understand the question properly.

Reply:
Yes, so my concern is: usually during the middle sections, the tries in the prefetcher won't be available, should we need to "try to fetch it from prefetcher"?

But yes, the cost is tiny and for most cases, we won't access trie since we have the snapshot.